tests/lib/metadata_linux_bridge_test.py

```diff
@@ -0,0 +1,365 @@
+
+
+BRIDGE0 = "br0"
+BRIDGE1 = "br1"
+IFACE0 = "eth0"
+IFACE1 = "eth1"
+
+
+class TestLinuxBridgeMetadata:
+    def test_creation_with_new_ports(self):
+        desired_state = state.State(
+            {
+                Interface.KEY: [
+                    _create_bridge_state(BRIDGE0, (IFACE0, IFACE1)),
+                    _create_iface_state(IFACE0, InterfaceType.UNKNOWN),
+                    _create_iface_state(IFACE1, InterfaceType.UNKNOWN),
+                ]
+            }
+        )
+        current_state = state.State({})
+        expected_dstate = state.State(
+            {
+                Interface.KEY: [
+                    _create_bridge_state(BRIDGE0, (IFACE0, IFACE1)),
+                    _create_iface_state(
+                        IFACE0,
+                        InterfaceType.UNKNOWN,
+                        extra=_create_metadata_state(BRIDGE0, IFACE0),
+                    ),
+                    _create_iface_state(
+                        IFACE1,
+                        InterfaceType.UNKNOWN,
+                        extra=_create_metadata_state(BRIDGE0, IFACE1),
+                    ),
+                ]
+            }
+        )
+        expected_cstate = state.State(current_state.state)
+
+        metadata.generate_ifaces_metadata(desired_state, current_state)
+
+        assert state.State(desired_state.state) == expected_dstate
+        assert current_state == expected_cstate
+
+    def test_creation_with_existing_ports(self):
+        desired_state = state.State(
+            {Interface.KEY: [_create_bridge_state(BRIDGE0, (IFACE0, IFACE1))]}
+        )
+        current_state = state.State(
+            {
+                Interface.KEY: [
+                    _create_iface_state(
+                        IFACE0, InterfaceType.UNKNOWN, InterfaceState.UP
+                    ),
+                    _create_iface_state(
+                        IFACE1, InterfaceType.UNKNOWN, InterfaceState.UP
+                    ),
+                ]
+            }
+        )
+        expected_dstate = state.State(
+            {
+                Interface.KEY: [
+                    _create_bridge_state(BRIDGE0, (IFACE0, IFACE1)),
+                    _create_iface_state(
+                        IFACE0,
+                        ifstate=InterfaceState.UP,
+                        extra=_create_metadata_state(BRIDGE0, IFACE0),
+                    ),
+                    _create_iface_state(
+                        IFACE1,
+                        ifstate=InterfaceState.UP,
+                        extra=_create_metadata_state(BRIDGE0, IFACE1),
+                    ),
+                ]
+            }
+        )
+        expected_cstate = state.State(current_state.state)
+
+        metadata.generate_ifaces_metadata(desired_state, current_state)
+
+        assert state.State(desired_state.state) == expected_dstate
+        assert current_state == expected_cstate
+
+    def test_editing_option(self):
```

Comment:
What is the option that is edited here?

Reply:
I need to understand what this one is... looks suspicious indeed.
It was copied from the OVS bridge.

Reply:
I changed the test name.